tez-dag/src/main/java/org/apache/tez/dag/app/dag/impl/TaskImpl.java

```diff
@@ -1043,6 +1043,10 @@ public void transition(TaskImpl task, TaskEvent event) {
         LOG.info("Ignore speculation scheduling since there is no running attempt on task {}.", task.getTaskId());
         return;
       }
+      if (task.commitAttempt != null) {
+        LOG.info("Ignore speculation scheduling since there is commitAttempt on task {}.", task.commitAttempt);
```

Comment:
LOG.info("Ignore speculation scheduling for task {} since commit has started with commitAttempt {}.", task.getTaskId, task.commitAttempt);

Reply:
done